### PR TITLE
cli: Support dumping master config

### PR DESCRIPTION
**Pull request by neilconway**
# Test Plan
- UI change:
  - [ ] add screenshots
  - [ ] React? build & check storybooks
- [ ] user-facing api change: modify documentation and examples
- [ ] user-facing api change: add the "User-facing API Change" label
- [ ] bug fix: add regression test
- [ ] bug fix: determine if there are other similar bugs in the codebase
- [ ] new feature: add test coverage for any user-facing aspects
- [ ] refactor: maintain existing code coverage

### CODE REVIEW

File: docs/reference/cluster-config.txt [hunk 1]
```diff
@@ -3,14 +3,18 @@
...
+configuration variables; this can be done using a configuration file,
+environment variables, or command-line options. (Although values from
+different sources will be merged, we generally recommend sticking to a
+single source for each service to keep things simple.)
+
+The master and the agent both accept an optional ``--config-file``
+command-line option, which specifies the path of the configuration file
+to use. Note that when running the master or agent inside a container,
+you will need to make the configuration file accessible inside the
+container (e.g., via a bind-mount). For example, this command starts the
```
Comment (dzhu) [inline docs/reference/cluster-config.txt:16]: "bind mount"

File: docs/reference/cluster-config.txt [hunk 2]
```diff
@@ -161,6 +165,9 @@ environment variables, or command-line options. The configuration file
 is normally located at ``/etc/determined/master.yaml`` on the master and
 is read when the master starts.
 
+The configuration of an active master can be examined using the
+Determined CLI, by running ``det master config``.
```
Comment (dzhu) [inline docs/reference/cluster-config.txt:169]: Perhaps "Determined CLI with the command \`\`det master config\`\`".

File: cli/determined_cli/master.py [hunk 3]
```diff
@@ -51,6 +57,7 @@ def process_response(response: Response, latest_log_id: int) -> int:
 
 args_description = [
     Cmd("m|aster", None, "manage master", [
+        Cmd("config", config, "fetch master config", []),
```
Comment (dzhu) [inline cli/determined_cli/master.py:60]: Maybe "...as JSON".

File: master/internal/core.go [hunk 4]
```diff
@@ -73,6 +73,10 @@ func New(version string, logStore *logger.LogBuffer, config *Config) *Master {
 	}
 }
 
+func (m *Master) getConfig(c echo.Context) (interface{}, error) {
+	return m.config.Printable()
```
Comment (ybt195) [inline master/internal/core.go:77]: non-blocking: Can we add this to the master info endpoint instead?

Comment (neilconway) [reply master/internal/core.go:77]: That would also work, of course -- why do you think it would be preferable?

Comment (ybt195) [reply master/internal/core.go:77]: It seems to fit that endpoint already and would be preferable to adding more endpoints.

Comment (neilconway) [reply master/internal/core.go:77]: Yeah, I thought about that, but I felt like (a) none of the existing clients of that endpoint care about this information (b) the client for this new endpoint doesn't care about the stuff in `/info` (c) the cost of a new endpoint is pretty minimal. So I'm inclined to just add a new endpoint.

**Pull request merged**